sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/BeamSqlEnv.java

```diff
@@ -133,6 +136,7 @@ private BeamSqlEnvBuilder(TableProvider tableProvider) {
       functionSet = new HashSet<>();
       autoLoadUdfs = false;
       autoLoadBuiltinFunctions = false;
+      pipelineOptions = PipelineOptionsFactory.create();
```

Comment:
I think we should force that pipeline options are set. I would set it to null here and add a `checkNotNull(pipelineOptions)` in the `build()` method